src/compiler/commandLineParser.ts

```diff
@@ -1346,7 +1346,7 @@ namespace ts {
             const categorizedOptions = createMultiMap<CommandLineOption>();
             for (const option of optionDeclarations) {
                 const { category } = option;
-                if (category !== undefined && category !== Diagnostics.Command_line_Options && category !== Diagnostics.Advanced_Options) {

```

Comment:
i do not think this the correct fix. we still do not want to write all options in the config file, it makes it too long and noisy. what you need is to add it if it were defined

Reply:
@mhegazy ok, will change